pkg/auth/aws/client.go

```diff
@@ -269,25 +272,61 @@ func Config(region, providerConfigRef *string, log logging.Logger) (cfg aws.Conf
 		log.Info("Using WebIdentity credentials")
 		stsclient := sts.NewFromConfig(cfg)
 
-		assumeRoleArn = &pcfg.AssumeRoleChain[0].RoleARN
-		log.Info("Assuming role", "role", *assumeRoleArn)
-		if cfg, err = config.LoadDefaultConfig(
-			ctx,
-			config.WithRegion(*region),
-			config.WithCredentialsProvider(aws.NewCredentialsCache(
-				stscredsv2.NewAssumeRoleProvider(
-					stsclient,
-					*assumeRoleArn,
-				)),
-			),
-		); err != nil {
-			err = errors.Wrapf(err, "failed to load aws config for assume role '%q'", *assumeRoleArn)
+		if len(pcfg.AssumeRoleChain) > 0 {
+			assumeRoleArn = &pcfg.AssumeRoleChain[0].RoleARN
+
+			log.Info("Assuming role", "role", *assumeRoleArn)
+			if cfg, err = config.LoadDefaultConfig(
```

Comment:
I wouldn't use named return values. But if we should, I don't think the `if` here is necessary (similar to line 301), as the error is already stored in `err`.

Reply:
the named return is really difficult for me to read too, I think I can refactor the library to use named returns pretty easy but it will "hide" the actual changes, do you think its fine to do it in this same PR?

Reply:
I think it's better to do it in a separate PR

Reply:
Will you "fix" the if on the other PR too?